src/python/pants/backend/python/util_rules/pex_requirements.py

```diff
@@ -47,15 +49,8 @@
 
 @dataclass(frozen=True)
 class Lockfile:
```

Comment:
I would love a docstring on the `Lockfile` type here. Esp describing how `url` works with regard to regular files vs resources.

It's currently not available from Pants cli help as it's embedded in the `LoadedLockfile`/`LoadedLockfileRequest` types.. but it wouldn't be too difficult to support drilling down into the fields of types in the future, which would allow to get this help with something like `pants LoadedLockfileRequest/lockfile --help`.

For reference, this works today:
```
pants LoadedLockfile --help

`pants.backend.python.util_rules.pex_requirements.LoadedLockfile` api type
--------------------------------------------------------------------------

A lockfile after loading and header stripping.

Validation is deferred until consumption time, because each consumed subset (in the case of a
PEX-native lockfile) can be individually validated.

activated by: pants.core

Include API types and rules dependency information by running `pants help-advanced pants.backend.python.util_rules.pex_requirements.LoadedLockfile`.
```

Reply:
Noted for a future change. I don't want to drive a new CI run just for this, esp since this is an internal type that as you say is not accessible via help today.